runtime/jcl/common/java_lang_Class.cpp

```diff
@@ -1818,6 +1824,144 @@ setNestmatesIncompatibleClassChangeError(J9VMThread *vmThread, J9Class *nestMemb
 	j9mem_free_memory(msg);
 }
 
+/**
+ * Loads, sets, verifies, & returns the claimed nest host of a class
+ *
+ * If the claimed nest host of a class is successfully loaded and verified,
+ * then the nest host field of the class is set and the nest host is returned.
+ * Otherwise, the nest host field is not set and the class itself is returned.
+ * If canThrow is true and loading/verifying the nest host fails, then an
+ * exception will be set.
+ *
+ * @param vmthread[in] the current vm thread
+ * @param clazz[in,out] class who's nest host is being acquired
+ * @param canThrow[in] true if a verification error/exception can be set & thrown
+ *
+ * @returns the nest host claimed by the class or the class itself
+ */
+static J9Class *
+loadAndCheckNestHost(J9VMThread *vmThread, J9Class *clazz, BOOLEAN canThrow) {
+	J9Class *nestHost = clazz->nestHost;
+
+	if (NULL == nestHost) {
+		J9JavaVM *vm = vmThread->javaVM;
+		J9InternalVMFunctions *vmFuncs = vm->internalVMFunctions;
+		PORT_ACCESS_FROM_VMC(vmThread);
+
+		J9ROMClass *romClass = clazz->romClass;
+		J9UTF8 *className = J9ROMCLASS_CLASSNAME(clazz->romClass);
+		J9UTF8 *nestHostName = J9ROMCLASS_NESTHOSTNAME(romClass);
+		BOOLEAN verified = FALSE;
+
+		char *nlsMsg = NULL;
+
+		/* If no nest host is named, class is own nest host */
+		if (NULL == nestHostName) {
+			nestHost = clazz;
+			verified = TRUE;
+		} else {
+			UDATA classLoadingFlags = 0;
+			if (TRUE == canThrow) {
+				classLoadingFlags = J9_FINDCLASS_FLAG_THROW_ON_FAIL;
+			} else {
+				classLoadingFlags = J9_FINDCLASS_FLAG_EXISTING_ONLY;
+			}
+			nestHost = vmFuncs->internalFindClassUTF8(vmThread, J9UTF8_DATA(nestHostName), J9UTF8_LENGTH(nestHostName), clazz->classLoader, classLoadingFlags);
+
+			/*
+			 * If this can throw, then we must check that the nest host is successfully
```

Comment:
fix comment:
`/* If this can throw, then we must check that the nest host is successfully`